### DISCUSSION TITLE
为什么每次使用xmake config -xxx=xxx时，除了命令行指定的配置以外，其它的配置都会变成默认配置？

### DISCUSSION THREAD

**Discussion by kokoro11** [xmake-io/xmake#4891] [Q&A] [answered]
比如说我的host环境是linux x86_64，我想将配置的plat改为mingw，arch改为x86，我设想的是输入这两条命令：
```
xmake f -p mingw
xmake f -a x86
```
但是每次运行完一条命令之后其它的配置都会变成默认配置，所以第一条命令执行完之后plat是mingw，arch是x86_64；第二条命令执行完之后plat是linux，arch是x86_64，这并不是我想要的结果。
我知道可以同时将plat和arch写在一条命令里，但是这样写有时候很麻烦，尤其是在项目的自定义配置变多了以后。
所以有没有什么办法可以只修改部分配置而让其它配置保持不变？

**Comment by star-hengxing** [comment, marked as answer]
相关讨论：https://github.com/xmake-io/xmake/issues/2401

可以考虑用社区贡献的插件

```sh
$ xmake plugin --install https://github.com/SirLynix/xmake-plugins
$ xmake cu -m debug
$ xmake cu -a x86
```

**Reply by kokoro11** [reply]
Edit：插件的owner合并了pr，可以继续使用原来的repo了。
这个插件有个bug，一次cu失败后就不能继续用cu了。目前的解决方法是把main.lua第41行的clean改成check。如果有人也遇到了这个bug，建议用我的fork：https://github.com/kokoro11/xmake-plugins

**Comment by waruqi** [comment]
不支持增量配置，这种设计的细节和实现复杂度太高，也更容易出问题，不会考虑支持。目前 autoconf/cmake 也都是全量配置。

**Reply by kokoro11** [reply]
啊这。。
